master/internal/trial.go

```diff
@@ -1015,3 +1007,17 @@ func (t *trial) trialClosing() bool {
 		(t.close != nil && t.sequencer.UpToDate()) ||
 		model.StoppingStates[t.experimentState]
 }
+
+func (t *trial) terminate(ctx *actor.Context, kill bool) {
+	if kill || !t.allReady(ctx){
+		ctx.Log().Info("killing trial")
+		t.killed = true
+		if t.task != nil {
+			ctx.Tell(t.rp, scheduler.TerminateTask{TaskID: t.task.ID, Forcible: true})
+		}
+	} else {
+		ctx.Log().Info("terminating trial")
+		t.pendingGracefulTermination = true
+		actors.NotifyAfter(ctx, time.Minute, terminateTimeout{runID: t.runID})
```

Comment:
blocking: this `terminateTimeout` shouldn't be triggered when a graceful termination starts. Related to my comment on L388.